tests/unit/source_code/test_dependencies.py

```diff
@@ -0,0 +1,238 @@
+from typing import BinaryIO
+from unittest.mock import create_autospec
+
+import pytest
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ObjectInfo, Language, ObjectType
+
+from databricks.labs.ucx.hive_metastore.migration_status import MigrationIndex
+from databricks.labs.ucx.source_code.dependencies import DependencyLoader
+from databricks.labs.ucx.source_code.languages import Languages
+from databricks.labs.ucx.source_code.notebook import Notebook
+from databricks.labs.ucx.source_code.source_migrator import SourceCodeMigrator
+from tests.unit import _load_sources
+
+
+def test_build_dependency_graph_visits_notebook_notebook_dependencies():
+    paths = ["root3.run.py.txt", "root1.run.py.txt", "leaf1.py.txt", "leaf2.py.txt"]
+    sources: dict[str, str] = dict(zip(paths, _load_sources(Notebook, *paths)))
+    visited: dict[str, bool] = {}
+
+    # can't remove **kwargs because it receives format=xxx
+    # pylint: disable=unused-argument
+    def download_side_effect(*args, **kwargs):
+        filename = args[0]
+        if filename.startswith('./'):
+            filename = filename[2:]
+        visited[filename] = True
+        result = create_autospec(BinaryIO)
+        result.__enter__.return_value.read.return_value = sources[filename].encode("utf-8")
+        return result
+
+    def get_status_side_effect(*args):
+        path = args[0]
+        return ObjectInfo(path=path, language=Language.PYTHON, object_type=ObjectType.NOTEBOOK)
+
+    ws = create_autospec(WorkspaceClient)
+    ws.workspace.download.side_effect = download_side_effect
+    ws.workspace.get_status.side_effect = get_status_side_effect
+    migrator = SourceCodeMigrator(ws, Languages(create_autospec(MigrationIndex)), DependencyLoader(ws))
```

Comment:
```suggestion
    migrator = SourceCodeMigrator(ws, Languages(empty_index), DependencyLoader(ws))
```
use `empty_index` fixture

Reply:
done